internal/manager/task_stash_box_tag.go

```diff
@@ -183,7 +210,261 @@ func (t *StashBoxPerformerTagTask) stashBoxPerformerTag(ctx context.Context) {
+			/*
+				if runParentCreateHook {
+					r.hookExecutor.ExecutePostHooks(ctx, *newStudio.ParentID, plugin.StudioCreatePost, input, nil)
+				} else if runParentUpdateHook {
+					r.hookExecutor.ExecutePostHooks(ctx, *newStudio.ParentID, plugin.StudioUpdatePost, input, parentTranslator.getFields())
+				}
+				r.hookExecutor.ExecutePostHooks(ctx, studioID, plugin.StudioCreatePost, input, nil)
+			*/
+		}
+	} else {
+		var name string
+		if t.name != nil {
+			name = *t.name
+		} else if t.studio != nil {
+			name = t.studio.Name
+		}
+		logger.Infof("No match found for %s", name)
+	}
+}
+
+// Duplicated in internal/identify/studio.go
+func studioFromScrapedStudio(ctx context.Context, input *models.ScrapedStudio, endpoint string, excluded map[string]bool) (*models.Studio, error) {
+	// Populate a new studio from the input
+	newStudio := models.Studio{
+		Name: input.Name,
+		StashIDs: models.NewRelatedStashIDs([]models.StashID{
+			{
+				Endpoint: endpoint,
+				StashID:  *input.RemoteSiteID,
+			},
+		}),
+	}
+
+	if input.URL != nil && !excluded["url"] {
+		newStudio.URL = *input.URL
+	}
+
+	if input.Parent != nil && input.Parent.StoredID != nil && !excluded["parent"] {
+		parentId, _ := strconv.Atoi(*input.Parent.StoredID)
+		newStudio.ParentID = &parentId
+	}
+
+	// Process the base 64 encoded image string
+	if input.Image != nil && !excluded["image"] {
+		var err error
+		newStudio.ImageBytes, err = utils.ProcessImageInput(ctx, *input.Image)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &newStudio, nil
+}
+
+// Duplicated in internal/identify/studio.go
+func studioPartialFromScrapedStudio(ctx context.Context, input *models.ScrapedStudio, id *string, endpoint string, excluded map[string]bool) (*models.StudioPartial, error) {
+	partial := models.NewStudioPartial()
+	partial.ID, _ = strconv.Atoi(*id)
+
+	if input.Name != "" && !excluded["name"] {
+		partial.Name = models.NewOptionalString(input.Name)
+
+	}
+
+	if input.URL != nil && !excluded["url"] {
+		partial.URL = models.NewOptionalString(*input.URL)
+	}
+
+	if input.Parent != nil && !excluded["parent"] {
+		if input.Parent.StoredID != nil {
+			parentID, _ := strconv.Atoi(*input.Parent.StoredID)
+			if parentID > 0 {
+				// This is to be set directly as we know it has a value and the translator won't have the field
+				partial.ParentID = models.NewOptionalInt(parentID)
+			}
+		}
+	} else {
+		partial.ParentID = models.NewOptionalIntPtr(nil)
+	}
+
+	// Process the base 64 encoded image string
+	if len(input.Images) > 0 && !excluded["image"] {
+		partial.ImageIncluded = true
+		var err error
+		partial.ImageBytes, err = utils.ProcessImageInput(ctx, input.Images[0])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	partial.StashIDs = &models.UpdateStashIDs{
+		StashIDs: []models.StashID{
+			{
+				Endpoint: endpoint,
+				StashID:  *input.RemoteSiteID,
+			},
+		},
+		Mode: models.RelationshipUpdateModeSet,
```

Comment:
This looks a bit ugly now because I first have to load the existing studio loaded in order to load the existing ids.

It feels like there should be a way to just add new/updated ids and then the sqlite UpdatePartial method knows it's doing a partial update and not to replace all ids.